verifier.go

```diff
@@ -96,3 +102,32 @@ func drainBody(b io.ReadCloser) (res io.ReadCloser, data []byte, err error) {
 	}
 	return io.NopCloser(&buf), buf.Bytes(), nil
 }
+
+// Helper function to validate the request signature with the given key
+func (v *RequestVerifier) ValidateSignature(r *http.Request, key ...string) bool {
+	sig, found := getSignatureHeader(r)
+	if !found {
+		return false
+	}
+	computedSignature := v.Checksum(r, key...)
+	return strings.ToLower(sig) == computedSignature
+}
+
+func (v *RequestVerifier) UpdateKey(newKey string) {
+	v.secretKey = func() string {
+		return newKey
+	}
+}
+
+func getSignatureHeader(r *http.Request) (string, bool) {
+	sig := r.Header.Get("X-Request-Signature")
+	if sig != "" {
+		return sig, true
+	}
+	sig = r.Header.Get("x-request-signature")
```

Comment:
Why is this necessary given that the HTTP headers according to RFC 2616 not not case-sensitive?

If that's the case we probably should bring back the configuration parameter for the header name.